compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -297,37 +295,28 @@ internal class BindingModuleGenerator(
         (ktProperty.initializer as? KtClassLiteralExpression)
           ?.firstChild
           ?.requireFqName(module)
+          ?.toClassReference(module)
```

Comment:
@vRallev let me know how you feel about this; I initially tried converting the whole method body to use `ClassReference`s and `PropertyReference`s but I did not see a clear way to replicate the 
```
(ktProperty.initializer as? KtClassLiteralExpression)
          ?.firstChild
          ?.requireFqName(module)
```

logic for descriptors.

Reply:
Add a TODO and we can maybe take a look together at it later.